server/schedulers/hot_region.go

```diff
@@ -201,33 +213,100 @@ func (h *balanceHotRegionsScheduler) dispatch(typ BalanceType, cluster opt.Clust
 	return nil
 }
 
-func (h *balanceHotRegionsScheduler) calcPendingInfluence(storeStat statistics.StoreHotPeersStat, pending map[uint64]Influence) statistics.StoreHotPeersStat {
+func (h *hotScheduler) calcPendingInfluence(storeStat statistics.StoreHotPeersStat, pending map[uint64]Influence) statistics.StoreHotPeersStat {
 	for id, stat := range storeStat {
 		stat.FutureBytesRate += pending[id].ByteRate
 	}
 	return storeStat
 }
 
-func (h *balanceHotRegionsScheduler) analyzeStoreLoad(storesStats *statistics.StoresStats) {
-	readFlowScoreInfos := NormalizeStoresStats(storesStats.GetStoresBytesReadStat())
-	writeFlowScoreInfos := NormalizeStoresStats(storesStats.GetStoresBytesWriteStat())
-	readFlowMean := MeanStoresStats(storesStats.GetStoresBytesReadStat())
-	writeFlowMean := MeanStoresStats(storesStats.GetStoresBytesWriteStat())
+func filterUnhealthyStore(cluster opt.Cluster, storeStatsMap map[uint64]float64) {
+	stores := cluster.GetStores()
+	for _, store := range stores {
+		if store.IsTombstone() ||
+			store.DownTime() > cluster.GetMaxStoreDownTime() {
+			delete(storeStatsMap, store.GetID())
+		}
+	}
+}
+
+func (h *hotScheduler) updateStatsByPendingOpInfo(storeStatsMap map[uint64]float64, balanceType BalanceType) {
+	for storeID := range storeStatsMap {
+		if balanceType == hotRead {
+			storeStatsMap[storeID] += h.readPendingSum[storeID].ByteRate
+		} else {
+			storeStatsMap[storeID] += h.writePendingSum[storeID].ByteRate
+		}
+	}
+
+	for regionID, pendingOpInfos := range h.pendingOpInfos {
+		var n *list.Element
+		for e := pendingOpInfos.Front(); e != nil; e = n {
+			opInfo := e.Value.(*pendingInfluence)
+			n = e.Next()
+			if opInfo.balanceType != balanceType {
```

Comment:
Why do we need to check this? I think this part of code is doing GC of `h.pendingOpInfos`, so ignoring the `balanceType` is ok.

Reply:
Good